### PR TITLE
[MST-626] Add more detail and consistency to log messages

### PR DESCRIPTION
**Pull request by bseverino**
**Description:**

Added more useful attributes to log messages, in a `key=value` format that is easier to extract, and reduced duplicate exception logs.

**JIRA:**

[MST-626](https://openedx.atlassian.net/browse/MST-626)

**Pre-Merge Checklist:**

- [x] Updated the version number in `edx_proctoring/__init__.py` and `package.json` if these changes are to be released.
- [x] Described your changes in `CHANGELOG.rst`
- [x] Confirmed Github reports all automated tests/checks are passing.
- [x] Approved by at least one additional reviewer.

**Post-Merge:**

- [ ] Create a tag matching the new version number.

### CODE REVIEW

File: edx_proctoring/backends/rest.py [hunk 1]
```diff
@@ -184,7 +184,11 @@ def register_exam_attempt(self, exam, context):
         payload['status'] = 'created'
         # attempt code isn't needed in this API
         payload.pop('attempt_code', False)
-        log.debug(u'Creating exam attempt for %r at %r', exam['external_id'], url)
+        log.debug(
+            'Creating exam attempt for exam_id = {exam_id} (external_id = {external_id}) at {url}'.format(
```
Comment (Zacharis278) [inline edx_proctoring/backends/rest.py:188]: Something worth confirming is if spunk can extract the field with spaces on either side of equals. You may have to reformat this as `key=val`

If you already know the spaces work, disregard this.

Comment (bseverino) [reply edx_proctoring/backends/rest.py:188]: Good call, I think you're right

File: edx_proctoring/backends/software_secure.py [hunk 2]
```diff
@@ -80,7 +80,7 @@ def register_exam_attempt(self, exam, context):
 
         if status not in [200, 201]:
             err_msg = (
-                u'Could not register attempt_code = {attempt_code}. '
+                u'Could not register attempt_code={attempt_code}. '
```
Comment (idegtiarov) [inline edx_proctoring/backends/software_secure.py:83]: One nit question: why `u` prefix was left? It seems that it could be removed here as well.

**Review by idegtiarov — COMMENTED** [review]
Other LGTM! Thanks.

**Pull request merged**